networkx/algorithms/centrality/betweenness.py

```diff
@@ -97,6 +97,12 @@ def betweenness_centrality(
     then we are only counting the paths in one direction. They are
     undirected paths but we are counting them in a directed way.
     To count them as undirected paths, each should count as half a path.
+    
+    This algorithm is not guaranteed to work if edge weights or demands
```

Comment:
```suggestion

    This algorithm is not guaranteed to be precise if edge weights or demands
```

Reply:
It might output very different results in some cases if sigma[w] is reset from a large value to 0.0 because of wrong evaluation of the unequality, so maybe "work" or "be correct" is better here.

Reply:
> is reset from a large value to 0.0

Overflow is more of an integer problem - exceeding the maximum value for floating points should result in `inf`, not "rolling over" to 0.0:

```python
>>> v = 1.7976931348623157e+308  # Maximum representable value in 64-bit float
>>> v * 2
inf
```

Now that doesn't mean it won't completely break betweenness (or other functionality in NetworkX), but I think floating point precision, particularly in comparisons, is a much more common culprit than "overflow".

Reply:
@rossbar I agree. But what I meant was the line in the code that explicitly resets the value to zero because of wrong evaluation in the if condition, I didn't mean overflow.

Reply:
That's [line 314](https://github.com/networkx/networkx/blob/50f91a7ef6ade995af1e9fb6fe54b6eb1f08a825/networkx/algorithms/centrality/betweenness.py#L314).  That indicates that sigma values are fragile. Why is it defined this way? Time to go back to the literature I guess.  @ibrh is there a better way to compute sigma?

Reply:
I've written my own implementation of Brande's algorithm but used `math.isclose(a, b)` instead of direct `==` which is how I noticed the difference in results, but I'm not sure of the consequences this has on the performance. If it doesn't slow the code down, and if there are no other reasons for why `math.isclose` wouldn't be used, then I would implement it that way in line [311](https://github.com/networkx/networkx/blob/50f91a7ef6ade995af1e9fb6fe54b6eb1f08a825/networkx/algorithms/centrality/betweenness.py#L311) and line [316](https://github.com/networkx/networkx/blob/50f91a7ef6ade995af1e9fb6fe54b6eb1f08a825/networkx/algorithms/centrality/betweenness.py#L316)

Reply:
Just to be clesr, you are saying that `==` can be replaced by `math.isclose` and `vw_dist < seen[w]` can be replaced by `(vw_dist < seen[w] and not math.isclose(vw_dist, seen[w]))`.
I'm not sure that the `<` inequality can be handled for all cases with the version I show here because roundoff could push vw_dist from less than to greater than seen[w]. Right?

Reply:
@dschult Yes I meant it the way you wrote it. It only evaluates to true if vw_dist is significantly smaller than seen[w], and otherwise it's false. Even if it changes from less than to greater than, it would change from being insignificantly smaller to being insignificantly larger, for which both cases are evaluated to false. If it was significantly smaller and changed to insignificantly smaller, in that case it would lead to a wrong result again. So ultimately, using `math.isclose` wouldn't guarantee correct results either, but it would give equal distances a more likely chance of occurring, which is a good thing if float distances are derived from integers and related, leading to a higher likelihood of equal distances and therefore demanding a little more tolerance when compared. Maybe it's not necessary to change the code because there isn't a perfect solution, but I think a warning comment is a good idea.